### ISSUE TITLE
[import/no-duplicates] False positive on multiple imports from submodules

### ISSUE THREAD

**Issue by SCasarotto** [import-js/eslint-plugin-import#2054]
I am seeing the following warning twice for `.../node_modules/firebase/empty-import.d.ts' imported multiple times  import/no-duplicates` for the following code:

```
import firebase from 'firebase/app';
import 'firebase/auth';
import 'firebase/database';
```

Interesting I am no seeing this warning if its simply
```
import firebase from 'firebase/app';
import 'firebase/database';
```

Additionally I saw this PR https://github.com/benmosher/eslint-plugin-import/pull/1612 where a comment is made about circling back on the second case. The fact that the second case isn't triggering a warning makes me think that maybe this is just a bug handling 3 and not 2 🤷‍♂️ .

**Comment by ljharb** [comment]
This seems like it's related to the typescript resolver.

**Comment by SCasarotto** [comment]
If it helps I am using `eslint-import-resolver-typescript` with the following settings:
```
'import/resolver': {
      typescript: {
        alwaysTryTypes: true,
      },
    },
```
If this is an issue in that lib, happy to move this issue. I am a bit new to these two libs so I am stilling trying to get my bearings on what owns what part.

**Comment by ljharb** [comment]
What I mean is, if all of the imports are resolving to the same d.ts file, then the import plugin will assume they're duplicate imports.

**Comment by murilocp** [comment]
I have the same problem, but with date-fns lib.

this is the code where I got the "import/no-duplicates" warning:

```
import ptBR from 'date-fns/locale/pt-BR';
import { addDays, format, getYear } from 'date-fns';
```
this is not correct, right?

**Comment by ljharb** [comment]
@murilocp you're correct, those are two different specifiers/modules and should not be warned on.

**Comment by the-ult** [comment]
> I have the same problem, but with date-fns lib.
> 
> this is the code where I got the "import/no-duplicates" warning:
> 
> ```
> import ptBR from 'date-fns/locale/pt-BR';
> import { addDays, format, getYear } from 'date-fns';
> ```
> 
> this is not correct, right?

This stil seems to be a problem.
Is this something which should be fixed in date-fns or in estlint-plugin-import?

**Comment by ljharb** [comment]
@the-ult here, which is why there’s a help wanted label.

**Comment by JounQin** [comment]
duplicate of #1479
